google/cloud/aiplatform/featurestore/featurestore.py

```diff
@@ -301,24 +302,253 @@ def list_entity_types(
 
     @base.optional_sync()
     def delete_entity_types(
-        self, entity_type_ids: List[str], sync: bool = True,
+        self, entity_type_ids: List[str] = None, force: bool = False, sync: bool = True,
     ) -> None:
-        """Deletes entity_type resources in this Featurestore given their entity_type IDs.
+        """Deletes entity_type resources in this Featurestore.
         WARNING: This deletion is permanent.
 
         Args:
             entity_type_ids (List[str]):
-                Required. The list of entity_type IDs to be deleted.
+                Optional. The list of entity_type IDs to be deleted. If entity_type_ids is not set,
+                all entityTypes in this Featurestore will be deleted.
```

Comment:
This is convenient, but sounds scary :). I wonder if we should do smt to prevent user accidentally delete their data. Probably we shouldn't add this option.

Reply:
I totally agree. I was hoping to get some feedback on this function. I can remove the delete all by default, and add an example in the docstring on how to get `entity_type_ids` from `list_entity_types()` to achieve the delete all functionality. If later there is more usage request on the delete all child resources, I can add a separate `delete_all_entity_types()`.